vectorstores/options.go

```diff
@@ -59,3 +60,13 @@ func WithDeduplicater(fn func(ctx context.Context, doc schema.Document) bool) Op
 		o.Deduplicater = fn
 	}
 }
+
+// WithReplacement returns an Option for setting the replacement flag that could be used
+// when adding documents. If set to true, the existing document with the **Metadata** will
+// be replaced with the new document. If set to false(default case), the new document will
+// be added to the store and the existing document will be kept.
+func WithReplacement(replacement bool) Option {
```

Comment:
I see this only implemented in `pgvector`. I would love to see all other stores implement this option, or atleast initially return a `no implemented` error.